### DISCUSSION TITLE
Problems importing helper_functions

### DISCUSSION THREAD

**Discussion by DanteNoguez** [mrdbourke/pytorch-deep-learning#94] [Q&A] [answered]
Hello, friends,

I'm getting an error when I run the code to import helper_functions (Chapter 2, video 71, minute 10:53:53)

Using the same code from the video notebook, I get the following error:

```helper_functions.py already exists, skipping download
  File "/content/helper_functions.py", line 8
    <!DOCTYPE html>
    ^
SyntaxError: invalid syntax
```

The file appears to be downloaded in the files tab, although I always got this error. Even so, I haven't been able to use any functions. Can somebody help me?

<img width="1048" alt="Captura de Pantalla 2022-08-10 a la(s) 20 00 08" src="https://user-images.githubusercontent.com/100301825/184047814-d524c255-a550-4af7-84cb-7536af6ef9dd.png">

**Comment by mrdbourke** [comment, marked as answer]
Hi @DanteNoguez,

Looks like you might be downloading the HTML version of the `helper_functions.py` file.

If you delete the file and try again but using the "raw" GitHub link: 

- raw (this should work):  https://raw.githubusercontent.com/mrdbourke/pytorch-deep-learning/main/helper_functions.py
- not raw (this will likely error):  https://github.com/mrdbourke/pytorch-deep-learning/blob/main/helper_functions.py

it should work.

For example, using the code from [02. PyTorch Neural Network Classification section 4](https://www.learnpytorch.io/02_pytorch_classification/): 

```python
import requests
from pathlib import Path 

# Download helper functions from Learn PyTorch repo (if not already downloaded)
if Path("helper_functions.py").is_file():
  print("helper_functions.py already exists, skipping download")
else:
  print("Downloading helper_functions.py")
  request = requests.get("https://raw.githubusercontent.com/mrdbourke/pytorch-deep-learning/main/helper_functions.py")
  with open("helper_functions.py", "wb") as f:
    f.write(request.content)

from helper_functions import plot_predictions, plot_decision_boundary
```

**Reply by DanteNoguez** [reply]
Oh! How silly of me, sorry. Thanks for the help, Daniel, I'm loving the course, it's wonderful!

**Reply by mrdbourke** [reply]
Glad to hear you fixed it! And thank you for the kind words, I'm glad you're enjoying the course :)
